### PR TITLE
remove AsynchronousLineReader 

### PR DESCRIPTION
**Pull request by akiezun**
Fixes https://github.com/samtools/htsjdk/issues/579 by deleting the AsynchronousLineReader class as agreed. I also deprecated LineReaderUtil and removed all its uses. Finally I created SynchronousLineReader and used it everywhere as needed.

@droazen can you review?

### CODE REVIEW

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java [hunk 1]
```diff
@@ -1,94 +1,53 @@
...
+ * @Deprecated use {@link SynchronousLineReader} directly.
  * @author mccowan
  */
+@Deprecated
 public class LineReaderUtil {
+    @Deprecated
     public enum LineReaderOption {
-        ASYNCHRONOUS, SYNCHRONOUS
+        ASYNCHRONOUS,
+        SYNCHRONOUS
```
Comment (droazen) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: Shouldn't you add a note that `ASYNCHRONOUS` here now has no effect?

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java [hunk 2]
```diff
@@ -1,94 +1,53 @@
...
     /**
-     * Like {@link #fromBufferedStream(java.io.InputStream, LineReaderUtil.LineReaderOption)}, but the synchronicity
-     * option is determined by {@link htsjdk.samtools.Defaults}: if asynchronous I/O is enabled, an asynchronous line reader will be
-     * returned.
+     * @Deprecated use <code>new SynchronousLineReader(stream);</code>
      */
+    @Deprecated
     public static LineReader fromBufferedStream(final InputStream stream) {
-        return fromBufferedStream(stream, Defaults.USE_ASYNC_IO_FOR_TRIBBLE ? LineReaderOption.ASYNCHRONOUS : LineReaderOption.SYNCHRONOUS);
+        return new SynchronousLineReader(stream);
```
Comment (droazen) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: With these changes, what is `Defaults.USE_ASYNC_IO_FOR_TRIBBLE` still hooked up to, if anything?

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java [hunk 3]
```diff
@@ -1,94 +1,53 @@
...
-        ASYNCHRONOUS, SYNCHRONOUS
+        ASYNCHRONOUS,
+        SYNCHRONOUS
     }
 
     /**
-     * Like {@link #fromBufferedStream(java.io.InputStream, LineReaderUtil.LineReaderOption)}, but the synchronicity
-     * option is determined by {@link htsjdk.samtools.Defaults}: if asynchronous I/O is enabled, an asynchronous line reader will be
-     * returned.
+     * @Deprecated use <code>new SynchronousLineReader(stream);</code>
```
Comment (droazen) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: Method should still have a description of what it does even if it's now `@Deprecated`

File: src/tests/java/htsjdk/tribble/readers/SynchronousLineReaderUnitTest.java [hunk 4]
```diff
@@ -12,11 +12,11 @@
 /**
  * @author mccowan
  */
-public class LineReaderUtilTest {
+public class SynchronousLineReaderUnitTest {
     @Test
     public void testLineReaderIterator() throws Exception {
         final File filePath = new File(TestUtils.DATA_DIR + "gwas/smallp.gwas");
-        final LineIterator lineIterator = new LineIteratorImpl(LineReaderUtil.fromBufferedStream(new PositionalBufferedStream(new FileInputStream(filePath))));
+        final LineIterator lineIterator = new LineIteratorImpl(new SynchronousLineReader(new PositionalBufferedStream(new FileInputStream(filePath))));
```
Comment (droazen) [inline src/tests/java/htsjdk/tribble/readers/SynchronousLineReaderUnitTest.java]: Add a test case for the other `SynchronousLineReader` constructor as well

**Comment by droazen** [comment]
Review complete, back to @akiezun, merge after addressing comments.

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java (hunk 1 above)
Comment (akiezun) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: done

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java (hunk 2 above)
Comment (akiezun) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: it is - writers still use it

File: src/java/htsjdk/tribble/readers/LineReaderUtil.java (hunk 3 above)
Comment (akiezun) [inline src/java/htsjdk/tribble/readers/LineReaderUtil.java]: done

File: src/tests/java/htsjdk/tribble/readers/SynchronousLineReaderUnitTest.java (hunk 4 above)
Comment (akiezun) [inline src/tests/java/htsjdk/tribble/readers/SynchronousLineReaderUnitTest.java]: done

**Pull request merged**